client/client-multi/client-cpp/impl/utils/ThreadPool.cpp

```diff
@@ -16,89 +16,94 @@
 
 #include "kaa/utils/ThreadPool.hpp"
 
-#include "kaa/logging/Log.hpp"
-#include "kaa/common/exception/KaaException.hpp"
+#include <chrono>
+#include <stdexcept>
 
+#include "kaa/common/exception/KaaException.hpp"
 
 namespace kaa {
 
 class Worker {
 public:
-    Worker(ThreadPool& threadPool) : threadPool_(threadPool) {}
-    void operator() ();
+    Worker(ThreadPool& threadPool)
+        : threadPool_(threadPool)
+    { }
 
-private:
-    ThreadPool& threadPool_;
-};
-
-void Worker::operator ()()
-{
-    KAA_MUTEX_UNIQUE_DECLARE(tasksLock, threadPool_.threadPoolGuard_);
-
-    while (threadPool_.isRun_) {
-        while (threadPool_.isRun_ && threadPool_.tasks_.empty() && !threadPool_.isPendingShutdown_) {
-            KAA_CONDITION_WAIT(threadPool_.onNewTask_, tasksLock);
-        }
+    void operator() ()
+    {
+        while (true) {
+            ThreadPoolTask task;
 
-        if (!threadPool_.isRun_ || (threadPool_.isPendingShutdown_ && threadPool_.tasks_.empty())) {
-            return;
-        }
+            {
+                KAA_MUTEX_UNIQUE_DECLARE(tasksLock, threadPool_.threadPoolGuard_);
 
-        auto task = threadPool_.tasks_.front();
-        threadPool_.tasks_.pop_front();
+                threadPool_.onNewTask_.wait(tasksLock,
+                        [this]{ return threadPool_.state_ == ThreadPool::State::STOPPED || !threadPool_.tasks_.empty(); });
 
-        KAA_UNLOCK(tasksLock);
+                if (threadPool_.state_ == ThreadPool::State::STOPPED && threadPool_.tasks_.empty()) {
+                    return;
+                }
 
-        try {
-            task();
-        } catch (...) {}
+                task = std::move(threadPool_.tasks_.front());
+                threadPool_.tasks_.pop_front();
+            }
 
-        KAA_LOCK(tasksLock);
+            try {
+                task();
+            } catch (...) {}
```

Comment:
Agree. But by design this class is not expected to have any logger. It is just like an utility class that may not have any relationships with Kaa C++ SDK.